src/main/java/software/amazon/smithy/lsp/SmithyInterface.java

```diff
@@ -27,56 +27,26 @@ private SmithyInterface() {
 
     }
 
-    /**
-     * @param path File of single-file model.
-     * @return Returns either an Exception, or the ValidatedResult of a model.
-     */
-    public static Either<Exception, ValidatedResult<Model>> readModel(File path) {
-
-        try {
-            return Either.forRight(Model.assembler().discoverModels().addImport(path.getAbsolutePath()).assemble());
-        } catch (Exception e) {
-            return Either.forLeft(e);
-        }
-    }
-
     /**
      * Reads the model in a specified file, adding external jars to model builder.
      *
-     * @param path         path to smithy file
+     * @param files        list of smithy files
      * @param externalJars set of external jars
      * @return either an exception encountered during model building, or the result
      *         of model building
      */
-    public static Either<Exception, ValidatedResult<Model>> readModel(File path, Set<String> externalJars) {
+    public static Either<Exception, ValidatedResult<Model>> readModel(Collection<File> files,
```

Comment:
Remove all versions apart from this generic one.

External jars don't need to be a separate argument (yet), but in the future will, because we need to load validators differently.